frontend/src/shared/components/select-users/select-users.component.ts

```diff
@@ -0,0 +1,144 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+import { ToastrService } from 'ngx-toastr';
+import { BoardService } from 'src/services/board.service';
+import {
+  BoardType,
+  BusinessRole,
+  EnumToArrayElement,
+  IBoard,
+  IUserCard,
+} from './Models';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { ValidationConstants } from 'src/entity-models/const-resources/validation-constraints';
+
+@Component({
+  selector: 'tasque-select-users',
+  templateUrl: './select-users.component.html',
+  styleUrls: ['./select-users.component.sass'],
+})
+export class SelectUsersComponent implements OnInit {
+  users$!: Observable<IUserCard[]>;
+  usersCount = 0;
+  roles: EnumToArrayElement[];
+  isLoading = true;
+  public userEmail = '';
+  public rowspan = 0;
+  public validationConstants = ValidationConstants;
+  public emailControl: FormControl;
+  public searchForm: FormGroup = new FormGroup({});
+
+  @Input()
+  public board: IBoard;
+
+  constructor(private service: BoardService, private toastr: ToastrService) {
+    this.roles = Object.keys(BusinessRole)
+      .filter((v) => isNaN(Number(v)))
+      .map((name) => {
+        return {
+          id: BusinessRole[name as keyof typeof BusinessRole],
+          name,
+        };
+      });
+
+    // board should be passed as a parameter
+    // ================================
+    const boardName = prompt('Please, enter a board name (supported values - "WithRoles", "WithoutRoles")') as string;
+    const getBoard = (): IBoard => {
+      switch (boardName) {
+        case 'WithRoles':
+          return {
+            id: 1,
+            type: BoardType.Organization,
+            hasRoles: true,
+            users: [],
+          };
+        case 'WithoutRoles':
+          return {
+            id: 1,
+            type: BoardType.Board,
+            hasRoles: false,
+            users: [],
+          };
+        default:
+          throw TypeError('Unknown name');
+      }
+    };
+    const board = getBoard();
+    const key = this.service.createKey(board);
+
+    if (localStorage.getItem(key)) {
+      this.board = JSON.parse(localStorage.getItem(key) as string);
+    } else {
+      this.board = board;
+    }
```

Comment:
Can you make it more like a component? So, it won't load data from localstorage or send tons of http requests

Reply:
These strings are here only for testing purpose. The board should always be provided to component, so this part would be deleted as soon as this component will be included anywhere.

Reply:
P.S. `Observable<any>` will be changed to `Observable<TModel>` as well